src/reflect/scala/reflect/internal/Types.scala

```diff
@@ -3598,6 +3602,14 @@ trait Types
     case tp :: Nil  => tp
     case _          => refinedType(tps, commonOwner(tps))
   }
+  def intersectionTypeForLazyBaseType(tps: List[Type]) = tps match {
+    case tp :: Nil  => tp
+    case _          => RefinedType(tps, newScope, tps.head.typeSymbolDirect)
+  }
+  def isIntersectionTypeForLazyBaseType(tp: RefinedType) = tp.parents match {
+    case head :: _ => tp.typeSymbolDirect eq head.typeSymbolDirect
```

Comment:
I'm also using the fact that we're using the underlying type symbol as the type symbol of the `RefinedType` as the way to distinguish them here.